modules/plot_circular_path.py

```diff
@@ -0,0 +1,95 @@
+"""
+Module to plot n circular waypoints, given a center and radius.
+"""
+
+import csv
+import math
+
+from modules.common.mavlink.modules.drone_odometry import DronePosition
+from modules.common.mavlink.modules.drone_odometry_local import DronePositionLocal
+from modules.common.mavlink.modules.local_global_conversion import drone_position_global_from_local
+
+from .waypoint import Waypoint
+
+
+def move_coordinates_by_offset(
+    start_point: Waypoint, offset_x: float, offset_y: float, name: str
+) -> Waypoint:
+    """Given a starting waypoint and offsets x and y displacements, find the
+    resulting waypoint.
+
+    Args:
+        starting_point (Waypoint): The starting waypoint
+        offset_x (float): The (potentially negative) displacement in west-east
+            direction, in meters
+        offset_y (float): The (potentially negative) displacement in north-south
+            direction, in meters
+        name (str): The name for the resulting waypoint
+
+    Returns:
+        Waypoint: the resulting waypoint after being moved by offset from the
+        original waypoint.
+    """
+    _, offset_local = DronePositionLocal.create(offset_y, offset_x, 0)
+
+    # Because drone_position_global_from_local requires DronePosition class,
+    # we need to convert Waypoint to DronePosition first
+    _, start_point_converted = DronePosition.create(
+        start_point.location_ground.latitude,
+        start_point.location_ground.longitude,
+        start_point.altitude,
+    )
+
+    _, end_point = drone_position_global_from_local(start_point_converted, offset_local)
+
+    return Waypoint(name, end_point.latitude, end_point.longitude, end_point.altitude)
+
+
+def generate_circular_path(center: Waypoint, radius: float, num_points: int) -> "list[Waypoint]":
```

Comment:
Let's have this return `"tuple[bool, list[Waypoint] | None]"`. E.g. what if someone inputs a negative radius or `num_points`? Have the first bool dictate whether this operation was successful or not.

Reply:
E.g. if we detect 0 or less num_points, return `false, None`